packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx

```diff
@@ -39,7 +68,22 @@ export function LinkMenu() {
     if (tuple) {
       const [node] = tuple
       setTitle((node.title as string) ?? '')
-      setURL((node.url as string) ?? '')
+
+      let nodeUrl = node.url as string
+
+      if (nodeUrl.startsWith(prefixType.https)) {
+        setPrefix(prefixType.https)
+        nodeUrl = nodeUrl.replace(prefixType.https, '')
+      } else if (nodeUrl.startsWith(prefixType.http)) {
+        setPrefix(prefixType.http)
+        nodeUrl = nodeUrl.replace(prefixType.http, '')
+      } else if (nodeUrl.startsWith(prefixType.mailto)) {
+        setPrefix(prefixType.mailto)
+        nodeUrl = nodeUrl.replace(prefixType.mailto, '')
+      } else {
+        setPrefix(prefixType.other)
+      }
+      setURL((nodeUrl as string) ?? '')
```

Comment:
you shouldn't need the code above till line 71 since you're setting the URL and then the `useEffect` above on line 45 should do all the magic.

Reply:
Ok but I still need to check for it being other because otherwise it defaults to http